libgadget/petapm.c

```diff
@@ -607,13 +619,18 @@ static void to_region(double * cell, double * region) {
     *region = *cell;
 }
 
-static void layout_build_and_exchange_cells_to_local(struct Layout * L) {
-    L->BufSend = mymalloc("PMBufSend", L->NcExport * sizeof(double));
+static void layout_build_and_exchange_cells_to_local(struct Layout * L, double * meshbuf, double * real) {
     L->BufRecv = mymalloc("PMBufRecv", L->NcImport * sizeof(double));
     int i;
     int offset;
 
-    layout_iterate_cells(L, to_region);
+    /*layout_iterate_cells transfers real to L->BufRecv*/
+    layout_iterate_cells(L, to_region, real);
+
+    /*Real is done now: reuse the memory for BufSend*/
+    myfree(real);
```

Comment:
Then we free real before allocating PMBufSend, so that PMBufSend can go into the memory previously allocated for it.